src/components/view/MapLocation.tsx

```diff
@@ -18,12 +19,27 @@ export function MapLocation (props: {row: number, column: number, gridInterval:
   const ModelComponent = tileDefinition.modelComponent
 
   const [hover, setHover] = useState(false)
+  const [mouseX, setMouseX] = useState(0)
+  const [mouseY, setMouseY] = useState(0)
   const onPointerOver = useCallback(() => setHover(true), [])
   const onPointerOut = useCallback(() => setHover(false), [])
-
-  const onClick = useCallback(() => {
-    props.onSelected()
-  }, [props])
+  const onPointerDown = useCallback((event) => {
+    if (!(event.button === LEFT_MOUSE_BUTTON)) {
+      return
+    }
+    setMouseX(event.clientX)
+    setMouseY(event.clientY)
+  }, [])
+  const onPointerUp = useCallback((event) => {
+    if (!(event.button === LEFT_MOUSE_BUTTON)) {
+      return
+    }
+    console.log('Mouse Button', event.button)
+    // Have a little fudge factor for mouse movement
+    if (Math.abs(event.clientX - mouseX + event.clientY - mouseY) < 50) {
+      props.onSelected()
+    }
+  }, [props, mouseX, mouseY])
```

Comment:
Fun fact: The reason exhaustive-deps requires the full `props` here is because `onSelected` is a function and the way it gets called means that its `this` value (the `props` object) should be preserved.